tests/test_project.py

```diff
@@ -245,3 +250,85 @@
 
     result = await client.create_environment(project_id=project_id, name="test_env2", environment_id=env_id)
     assert result.code == 500
+
+
+@pytest.mark.asyncio
+async def test_environment_listener(server, client_v2):
+    class EnvironmentListenerCounter(EnvironmentListener):
+        def __init__(self):
+            self.counter = 0
+
+        async def environment_action(self, action: EnvironmentAction, env: model.Environment) -> None:
+            self.counter += 1
+
+    create_environment_listener = EnvironmentListenerCounter()
+    update_environment_listener = EnvironmentListenerCounter()
+    delete_environment_listener = EnvironmentListenerCounter()
+    clear_environment_listener = EnvironmentListenerCounter()
+
+    environment_service = cast(EnvironmentService, server.get_slice(SLICE_ENVIRONMENT))
+    environment_service.register_listener(EnvironmentAction.created, create_environment_listener)
```

Comment:
Usually you want to register to multiple actions. Maybe add the possibility to register to a list of actions.